tool/util.py

```diff
@@ -0,0 +1,79 @@
+# Lint as: python3
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""utility functions used to build the property-value tree structure """
+import stat_config_pb2
+import collections
+from google.protobuf import text_format
+
+class PopObsSpec(object):
+  """Represents a StatisticalPopulation & Observation spec."""
+  def __init__(self, pop_type, mprop, stats, ps, cpv, text):
+      self.pop_type = pop_type #population type, string
+      self.mprop = mprop #measured property, string
+      self.stats = stats #stat_type, string
+      self.ps = ps #properties, list of string
+      self.cpv = cpv #constrained property-value pairs, dict{string: string}
+      self.text = text #name, string
+      self.key = (pop_type, mprop, stats) + tuple(sorted(ps + list(cpv.keys())))
+
+  def __str__(self):
```

Comment:
It's best to use json_dumps to generate json. It'll be easy to make a mistake and generate invalid json by doing string manipulation.

Reply:
Got it. Removed this function.